include/options.hxx

```diff
@@ -420,7 +424,7 @@ public:
                   << ")" << std::endl;
       return def;
     }
-    T val = as<T>(bout::utils::meshFromValue(def));
+    T val = as<T>(std::forward<T>(def));
```

Comment:
`def` is taken by value, so you can just pass it straight to `as`

Reply:
Passing it straight through caused problems when passing in strings; I got compiler errors about converting to `&&`. Possibly converting literal char strings to std::string, then passing. Probably changing to `const &` will fix this.

Reply:
Compiler error
```
../../include/options.hxx:427:7: error: no matching function for call to ‘Options::as<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::__cxx11::basic_string<char>&)’
     T val = as<T>(def);
       ^~~
../../include/options.hxx:359:5: note: candidate: ‘template<class T> T Options::as(T&&) const’
   T as(T&& UNUSED(similar_to) = {}) const {
     ^~
../../include/options.hxx:359:5: note:   template argument deduction/substitution failed:
../../include/options.hxx:427:19: note:   cannot convert ‘def’ (type ‘std::__cxx11::basic_string<char>’) to type ‘std::__cxx11::basic_string<char>&&’
     T val = as<T>(def);
```